QueAns/admin.py

```diff
@@ -0,0 +1,171 @@
+from django.contrib import admin
+from .models import Question, Meetings, Voting, Satisfaction
+from django.utils.safestring import mark_safe
+
+class MeetingsAdmin(admin.ModelAdmin):
+    list_display = ['meetings_text']
+
+    def get_readonly_fields(self, request, obj=None):
+        if not request.user.is_superuser:
+            if request.user == obj.moderator:
+                return ['moderator']
+            else:
+                return ['id', 'meetings_text', 'venue', 'moderator', 'start_time', 'end_time']
+        else:
+            return []
+
+    def has_add_permission(self, request):
+        if request.user.is_superuser:
+            return True
+
+    def has_delete_permission(self, request, obj=None):
+        if request.user.is_superuser:
+            return True
+        return False
+
+
+class QuestionAdmin(admin.ModelAdmin):
+    fields = ['question_text', 'anon_status', 'meeting_id']
+    list_filter = ['meeting_id', ]
+    list_display = ['question_text', 'upvote_button', 'upvote_count', 'downvote_button', 'downvote_count',
+                    'satisfied_button', 'satisfaction_count', 'disatisfied_button', 'disatisfaction_count']
+
+
+    def upvote_count(self, obj):
+        list = Voting.objects.filter(question_id=obj.id).values_list('vote')
+        list = list.filter(vote=1)
```

Comment:
This would mean this this will get called on each item in list display i.e. of there are 20 items there will be 20 queries to DB to fetch this.

Also, this is just upvote_count, then there will be queries to fetch downvote and satisfaction/dis count too. Making it 20x4 = 80 queries. Which is too much and might bloat DB.

Please read about select_related and prefecth_related and see if we can use any of it here.